CMEPS-interface/CMakeLists.txt

```diff
@@ -62,8 +69,103 @@ list(APPEND _mediator_files
   CMEPS/mediator/med_phases_post_rof_mod.F90
   CMEPS/mediator/med_phases_post_wav_mod.F90)
 
-### Create target library and set PUBLIC interfaces on the library
-add_library(cmeps STATIC ${_nems_util_files} ${_mediator_files})
+###############################################################################
+### Host model for CCPP
+###############################################################################
+
+### Query FV3 path
+get_filename_component(FV3_PATH "../FV3" REALPATH)
+
+### If FV3 is available build CCPP host model under CMEPS too
+if(EXISTS ${FV3_PATH})
+  ### Print out information
+  message("FV3_PATH = ${FV3_PATH}")
+  message("Compiling CMEPS with CCPP support.")
+
+  ### Set environment variable for FV3 directory, used by config file
+  set(ENV{FV3_PATH} ${FV3_PATH})
+
+  ### Look for CMEPS CCPP suites and get list of them to provide CCPP preprocessor
+  set(_ccpp_suites_arg "--suites=")
+  FILE(GLOB FILES "CMEPS/ufs/ccpp/suites/*.xml")
+  foreach(MYFILE ${FILES})
+    get_filename_component(TMPVAR ${MYFILE} NAME_WE)
+    string(REGEX REPLACE "suite_" "" CCPP_SUITE ${TMPVAR})
+    set(_ccpp_suites_arg "${_ccpp_suites_arg}${CCPP_SUITE},")
+  endforeach()
+  string(REGEX REPLACE ",$" "" _ccpp_suites_arg ${_ccpp_suites_arg})
+
+  ### Set options for ccpp_prebuild.py
+  if(DEBUG)
+    ### Enable debugging features in auto-generated physics caps
+    set(_ccpp_debug_arg "--debug")
+    ### Enable verbose output from ccpp_prebuild.py
+    set(_ccpp_verbose_arg "--verbose")
+  endif()
+
+  ### Trigger ccpp_prebuild.py script to connect host model with CCPP physics
+  message("Calling CCPP code generator (ccpp_prebuild.py) for suites ${_ccpp_suites_arg} ...")
+  execute_process(
+                  COMMAND ${Python_EXECUTABLE} 
+                  "${FV3_PATH}/ccpp/framework/scripts/ccpp_prebuild.py"
+                  "--config=CMEPS/ufs/ccpp/config/ccpp_prebuild_config.py"
+                  "--builddir=${CMAKE_CURRENT_BINARY_DIR}"
+                  "--namespace=med"
```

Comment:
Great to have this option.

Reply:
@climbfuji Thanks for your help to implement it.